### ISSUE TITLE
cvt_intrvl_str_to_us needs more helpful error message

### ISSUE THREAD

**Issue by baallan** [ovis-hpc/maestro#27]
The error message from cvt_intrvl_str_to_us is unhelpful to the user regarding what to fix in the input.
This is in general tricky (or verbose in source code terms) to solve with try/except syntax.
Generally, we don't have file location information from the parse tree (or do we???) that could be displayed when a time interval input error is raised.

A cheap remedy (though less that perfectly user-friendly) to the problem would be to change the input to cvt_intrvl_str_to_us from the string to be parsed to the map object and the key. This would be a minor adjustment to each cvt call site (commas vs []). It would allow the exception message to name the key (interval, reconnect, etc) which would probably be enough for an admin to find an offending line.
If the map object from yaml actually retains parse location info, that would be better than the key name to use in an error message.

A friendlier but more work remedy would be to add a context name argument to cvt argument list. Composing that string before calling cvt_intrvl_str_to_us  would add a line of code at each call site. The context would be the full yaml/etcd path/key name.

**Comment by nick-enoent** [comment]
@baallan Error messages and parsing were added to this function in commit 932b5c20b00fc617a2787b6de40f6df92d07af50 As far as using unitless integers, maybe we could discuss this at the users group?

**Comment by baallan** [comment]
I think Tom made the case at UG for having times units always, which means only strings should be supported. But there's a corner case I'd like to see supported: "0" should not require units.

The problem we have is that the python yaml parser is inconsistent in handling stuff. e.g.
1ms (unquoted) --> int(1000) while 1ns (unquoted) --> string(1ns) and more bizarrely some strings with : in them resolve to ints derived in obscure ways from hex time.  Some of the insanity is part of yaml 1.1 but not yaml 1.2. Some of the insanity is in pyyaml (libyaml) extensions of the yaml 1.1 spec.

The only reliable way forward to me seems to be for us to document as:

```
interval, offset, reconnect always require *quoted* strings that start with a decimal digit string, 
followed by a unit string and do not contain any other punctuation. For 0, units may be omitted.
```
This rules out float numbers, which is a good thing because we don't have to parse scientific notation or deal with internationalization of dot vs comma.

**Comment by baallan** [comment]
@nick-enoent the new time parsing fails on the cl-ex example where the old one did not.
Also, there should not be any automatic morphing of offset. if it's out of the half range, issue an error please.

**Comment by nick-enoent** [comment]
@baallan The time parsing has been fixed in pull request #34.
